test/framework/filetools.py

```diff
@@ -793,6 +793,12 @@ def test_apply_patch(self):
         pattern = "I'm a toy, and very proud of it"
         self.assertTrue(pattern in patched)
 
+        toy_patch_gz = os.path.join(testdir, 'sandbox', 'sources', 'toy', 'toy-0.0_gzip.patch.gz')
+        self.assertTrue(ft.apply_patch(toy_patch_gz, path))
```

Comment:
Yeah I patch the same file again as I figured that would be sufficient. We could do it the other way too. Mine also tests a dependent patch FWIW.